### ISSUE TITLE
MobileDatePicker does not have an icon

### ISSUE THREAD

**Issue by c3nj1** [mui/mui-x#4836]

**Comment by alexfauquette** [comment]
If you mean adding an icon in the `TextField` that is not directly related to pickers.

The date picker passes parameters to the TextField, but you can add your own to customize it, to use `TextFieldProps`

What you are looking for is [Input Adornments](https://mui.com/material-ui/react-text-field/#input-adornments)

**Comment by flaviendelangle** [comment]
We do add some icons by default to the other pickers.

Do we have a reason not to do it for `DateTimePicker` ?

![image](https://user-images.githubusercontent.com/3309670/167830292-4529ed1a-532a-4b16-8bc2-b0e7b648dd57.png)

**Comment by alexfauquette** [comment]
The icon is here to allow opening the views when it is not the default behavior

The same demo but with mobile emulation remove the icons for the responsive components

![image](https://user-images.githubusercontent.com/45398769/167831754-b608679a-0bf3-4002-8b45-b6277137021b.png)

**Comment by flaviendelangle** [comment]
Oh my bad

**Comment by c3nj1** [comment]
> If you mean adding an icon in the `TextField` that is not directly related to pickers.
> 
> The date picker passes parameters to the TextField, but you can add your own to customize it, to use `TextFieldProps`
> 
> What you are looking for is [Input Adornments](https://mui.com/material-ui/react-text-field/#input-adornments)

I mean, like @alexfauquette already mentioned, the icons used to opening the views. There are no icons in mobile view, but it would be nice to have them there as well.

**Comment by alexfauquette** [comment]
@c3nj1 I doubt it is a good idea to provide icons that are sometimes buttons opening views, and sometimes just a decoration.

If you want to add icons to provide visual information about the field, you have the possibility. Here is an example

https://codesandbox.io/s/materialuipickers-material-demo-forked-rwu3n1?file=/demo.tsx

**Comment by c3nj1** [comment]
@alexfauquette The icon should open the view - that's the primary case. But it should also be a custom icon, because of visualizing the meaning of the current Date. So the easiest way would be allowing the mobile version to have those icons as well. Or: what exactly speaks against allowing it in the mobile version as well?

**Comment by flaviendelangle** [comment]
Side note, it relates somehow with https://github.com/mui/mui-x/issues/4845#issuecomment-1123615995 
For SSR, the less diff we have between desktop and mobile (with view closed), the better

**Comment by alexfauquette** [comment]
What speaks against it, is adding a button in the DOM that has the same effect as the input associated. You add a button for new behavior, not for providing an information

I do not understand which pain points it will solve to have this icon set internally. Below are the code to add an icon when there is no icon by default, and the code to override an existing one. They do not look that much different

```jsx
<MobileDatePicker
  value={value}
  onChange={handleChange}
  InputProps={{
    endAdornment: (
      <InputAdornment position="end">
        <SaveIcon />
      </InputAdornment>
    )
  }}
  renderInput={(params) => <TextField {...params} />}
/>
```

```jsx
<MobileDatePicker
  value={value}
  onChange={handleChange}
  components={{
	OpenPickerIcon: SaveIcon
  }}
  renderInput={(params) => <TextField {...params} />}
/>
```

**Comment by c3nj1** [comment]
Ok, thank you all for your Support. We'll try it wit @alexfauquette recommendation for now 👍

**Comment by eggplants** [comment]
If you use DatePicker, this is probably the reason why the icon appears on desktop and not on mobile.

**Comment by LukasTy** [comment]
@eggplants Yeah, your provided import statement is not the most elegant as we didn't have this need to expose these icons yet.
I essence, those icons are just copies from `icons-material`, probably to reduce the bundle size not needing to add `icons-material` as a direct dependency to our module.
If you already have `@mui/icons-material` as a project dependency, then just using same icons from that package seems like a no-brainer. 😃 
```ts
import Event from '@mui/icons-material/Event';
```

**Comment by flaviendelangle** [comment]
> those icons are just copies from icons-material, probably to reduce the bundle size not needing to add icons-material as a direct dependency to our module.

Exactly

**Comment by eggplants** [comment]
I wish `DatePicker` had the prop like `showOpenPickerIconInMobile={true}` ...

**Comment by Hassaan0** [comment]
@mui/x-date-pickers doesn't provides a default icon / or any property to show/hide icon for MobileDatePicker. So, if you want to add a icon in MobileDatePicker you can try the below code. I tried to override the MuiInput feild icon so maybe it is not the best option but it can help for now 

Dependencies : 

- "moment": "^2.29.4"
- "@mui/styled-engine-sc": "^5.11.11"
- "@mui/styles": "^5.11.13"
- "@mui/x-date-pickers": "^6.2.1"

    ```
<MobileDatePicker
        value={moment(value)}
        format="DD MMM YYYY"
        slotProps={{
          textField: {
            style: {
              background: "white",
            },
            inputProps: {
              className: textFeildClasses,
            },
            InputProps: {
              endAdornment: (
                <InputAdornment position="end">
                  <CalendarViewDayOutlined />
                </InputAdornment>
              ),
            },
          },
        }}
      />
```

**Comment by fliodhais** [comment]
> This issue is probably the reason why the icon appears on desktop and not on mobile when I use `DatePicker`.
> 
> It can fix like this: https://stackblitz.com/edit/react-tl9pgs-kojw1y?file=demo.tsx
> 
> But the import statement is pretty weird.
> 
> ```ts
> import { Calendar } from '@mui/x-date-pickers/internals/components/icons';
> ```
> 
> Any workaround?

@Hassaan0 works
